src/components/tls-checker/index.ts

```diff
@@ -24,15 +24,21 @@
 
 import sslChecker from 'ssl-checker'
 
-export async function checkTLS(url: string, expiryThreshold = 30) {
-  const { valid, validTo, daysRemaining } = await sslChecker(url)
+export async function checkTLS(host: any, expiryThreshold = 30) {
+  const hostOptions = host?.options! ?? {}
+  const domain = host?.domain! ?? host
+
```

Comment:
I think it's easier to understand if we use the exact type for the `host` argument which is either a `string`, or an object with `domain` and `options` properties:

```typescript
type TLSHostArg = {
  domain: string
  options?: Record<string, any>
}
export async function checkTLS(
  host: string | TLSHostArg,
  expiryThreshold = 30
) {
  const hostOptions = (host as TLSHostArg)?.options ?? {}
  const domain = (host as TLSHostArg)?.domain ?? (host as string)
```

Reply:
I'm not seasoned with TypeScript, so I'm not familiar with the ramifications of that. If that would work here, go for it.

Reply:
Done